File: merge-tonic-lib/tonic/hydration/llm_config.py
"""Hydration LLM configuration resolved from environment variables."""

from __future__ import annotations

from dataclasses import dataclass
import os
from typing import Mapping


HydrationLlmMode = str


@dataclass(frozen=True)
class HydrationLlmConfig:
    mode: HydrationLlmMode
    model: str
    base_url: str
    api_key: str
    timeout_ms: int


def _normalize_mode(raw: str | None) -> HydrationLlmMode:
    value = (raw or "deterministic").strip().lower()
    if value == "openai":
        return "openai"
    return "deterministic"


def resolve_hydration_llm_config(env: Mapping[str, str] | None = None) -> HydrationLlmConfig:
    source = env or os.environ
    timeout_raw = str(source.get("TONIC_HYDRATION_LLM_TIMEOUT_MS", "")).strip()
    try:
        timeout_ms = int(timeout_raw)
    except ValueError:
        timeout_ms = 0
    return HydrationLlmConfig(
        mode=_normalize_mode(source.get("TONIC_HYDRATION_LLM_MODE")),
        model=str(source.get("TONIC_HYDRATION_LLM_MODEL", "gpt-4o-mini")).strip(),
        base_url=str(source.get("TONIC_HYDRATION_LLM_BASE_URL", "https://api.openai.com/v1")).strip().rstrip("/"),
        api_key=str(source.get("TONIC_HYDRATION_LLM_API_KEY") or source.get("OPENAI_API_KEY") or "").strip(),
        timeout_ms=timeout_ms if timeout_ms > 0 else 30_000,
    )

File: packages/tonic-core/src/hydration/llmConfig.ts
export type HydrationLlmMode = "deterministic" | "openai";

export type HydrationLlmConfig = {
  mode: HydrationLlmMode;
  model: string;
  baseUrl: string;
  apiKey: string;
  timeoutMs: number;
};

function normalizeMode(raw: string | undefined): HydrationLlmMode {
  const value = (raw ?? "deterministic").trim().toLowerCase();
  if (value === "openai") {
    return "openai";
  }
  return "deterministic";
}

export function resolveHydrationLlmConfig(
  env: NodeJS.ProcessEnv = process.env,
): HydrationLlmConfig {
  const timeoutRaw = Number.parseInt((env.TONIC_HYDRATION_LLM_TIMEOUT_MS ?? "").trim(), 10);
  return {
    mode: normalizeMode(env.TONIC_HYDRATION_LLM_MODE),
    model: (env.TONIC_HYDRATION_LLM_MODEL ?? "gpt-4o-mini").trim(),
    baseUrl: (env.TONIC_HYDRATION_LLM_BASE_URL ?? "https://api.openai.com/v1").trim().replace(/\/+$/, ""),
    apiKey: (env.TONIC_HYDRATION_LLM_API_KEY ?? env.OPENAI_API_KEY ?? "").trim(),
    timeoutMs: Number.isFinite(timeoutRaw) && timeoutRaw > 0 ? timeoutRaw : 30_000,
  };
}

